File: mobile/lib/logic/catalog_taxonomy.dart
import 'package:flutter/material.dart';

enum CatalogCategoryStatus { active, soon }

class CatalogCategory {
  const CatalogCategory({
    required this.id,
    required this.label,
    required this.description,
    required this.icon,
    required this.status,
    this.count,
  });

  final String id;
  final String label;
  final String description;
  final IconData icon;
  final CatalogCategoryStatus status;
  final int? count;
}

const catalogCategories = [
  CatalogCategory(
    id: 'telefon',
    label: 'Telefon',
    description: 'Akıllı telefon · teknik puan · kıyas',
    icon: Icons.smartphone_rounded,
    status: CatalogCategoryStatus.active,
  ),
  CatalogCategory(
    id: 'tablet',
    label: 'Tablet',
    description: 'iPad, Galaxy Tab ve diğer tabletler',
    icon: Icons.tablet_rounded,
    status: CatalogCategoryStatus.soon,
  ),
  CatalogCategory(
    id: 'bilgisayar',
    label: 'Bilgisayar',
    description: 'Dizüstü, masaüstü, mini PC',
    icon: Icons.laptop_mac_rounded,
    status: CatalogCategoryStatus.soon,
  ),
  CatalogCategory(
    id: 'akilli-saat',
    label: 'Akıllı saat',
    description: 'Apple Watch, Galaxy Watch',
    icon: Icons.watch_rounded,
    status: CatalogCategoryStatus.soon,
  ),
  CatalogCategory(
    id: 'kulaklik',
    label: 'Kulaklık',
    description: 'Kablosuz ve kablolu',
    icon: Icons.headphones_rounded,
    status: CatalogCategoryStatus.soon,
  ),
  CatalogCategory(
    id: 'tv',
    label: 'Televizyon',
    description: 'Smart TV',
    icon: Icons.tv_rounded,
    status: CatalogCategoryStatus.soon,
  ),
];

CatalogCategory catalogCategoryById(String id) {
  return catalogCategories.firstWhere((c) => c.id == id, orElse: () => catalogCategories.first);
}

File: src/lib/catalog-taxonomy.ts
export type CategoryStatus = "active" | "soon";

export interface ProductCategory {
  id: string;
  slug: string;
  label: string;
  shortLabel: string;
  description: string;
  status: CategoryStatus;
  /** Product count when active; omit for upcoming categories */
  count?: number;
}

export const productCategories: ProductCategory[] = [
  {
    id: "telefon",
    slug: "telefon",
    label: "Telefon",
    shortLabel: "Telefon",
    description: "Akıllı telefonlar · teknik puan · kıyas · kullanıcı oyu",
    status: "active",
  },
  {
    id: "tablet",
    slug: "tablet",
    label: "Tablet",
    shortLabel: "Tablet",
    description: "iPad, Galaxy Tab ve diğer tabletler",
    status: "soon",
  },
  {
    id: "bilgisayar",
    slug: "bilgisayar",
    label: "Bilgisayar",
    shortLabel: "PC",
    description: "Dizüstü, masaüstü ve mini PC",
    status: "soon",
  },
  {
    id: "akilli-saat",
    slug: "akilli-saat",
    label: "Akıllı saat",
    shortLabel: "Saat",
    description: "Apple Watch, Galaxy Watch ve benzeri",
    status: "soon",
  },
  {
    id: "kulaklik",
    slug: "kulaklik",
    label: "Kulaklık",
    shortLabel: "Kulaklık",
    description: "Kablosuz ve kablolu kulaklıklar",
    status: "soon",
  },
  {
    id: "televizyon",
    slug: "televizyon",
    label: "Televizyon",
    shortLabel: "TV",
    description: "Smart TV ve monitör-TV",
    status: "soon",
  },
];

export function getCategory(slug: string) {
  return productCategories.find((item) => item.slug === slug);
}

export function categoryHref(slug: string) {
  return `/katalog/${slug}`;
}
